docs/user/debugging.md

```diff
@@ -71,28 +79,132 @@ More information can be found in the [View Logs with Logcat][] help article.
+
+```Swift
+func application(_: UIApplication,
+                  open url: URL,
+                  options _: [UIApplication.OpenURLOptionsKey: Any] = [:]) -> Bool {
+    // This does nothing if the url isn't meant for Glean.
+    Glean.shared.handleCustomUrl(url: url)
+
+    return true
+}
+```
+
+If you need additional help setting up a custom URL scheme in your application, please refer to [Apple's documentation](https://developer.apple.com/documentation/uikit/inter-process_communication/allowing_apps_and_websites_to_link_to_your_content/defining_a_custom_url_scheme_for_your_app).
+
+### Invoking the Glean-iOS debug commands
+
+Now that the app has the Glean SDK debug functionality enabled, there are a few ways in which we can invoke the debug commands.
+
+#### Using a web browser
+
+Perhaps the simplest way to invoke the Glean SDK debug functionality is to open a web browser and type/paste the custom URL into the address bar.  This is especially useful on an actual device because there isn't a good way to launch from the command line and process the URL for an actual device.
+
+Using the glean-sample-app as an example: to activate ping logging, tag the pings to go to the Glean Debug View, and force the `events` ping to be sent, enter the following URL in a web browser on the iOS device:
+
+```shell
+glean-sample-app://glean?logPings=true&tagPings=My-ping-tag&sendPing=events
+```
+
+This should cause iOS to prompt you with a dialog asking if you want to open the URL in the Glean Sample App, and if you select "Okay" then it will launch (or resume if it's already running) the application with the indicated commands and parameters and immediately force the collection and submission of the events ping.
+
+> **Note:** This method does not work if the browser you are using to input the command is the same application you are attempting to pass the Glean debug commands to.  So, you couldn't use Firefox for iOS to trigger commands within Firefox for iOS.
+
+It is also possible to encode the URL into a 2d barcode and launch the app via the camera app.  After scanning the encoded URL, the dialog prompting to launch the app should appear as if the URL were entered into the browser address bar.
+
+#### Using the command line
+
+This method is useful for testing via the Simulator, which typically requires a Mac with Xcode installed, including the Xcode command line tools.  In order to perform the same command as above with using the browser to input the URL, you can use the following command in the command line terminal of the Mac:
+
+```shell
+xcrun simctl openurl booted "glean-sample-app://glean?logPings=true&tagPings=My-ping-tag&sendPing=events"
+```
+
+This will launch the simulator and again prompt the user with a dialog box asking if you want to open the URL in the Glean Sample App (or whichever app you are instrumenting and testing).
+
+#### Triggering commands if the app is already running
```

Comment:
I think this paragraph can be safely removed if we restructure this document, since this applies to all platforms and it's outlining the workflows.